sandpack-react/src/contexts/sandpackContext.tsx

```diff
@@ -389,6 +391,45 @@ class SandpackProvider extends React.PureComponent<
     });
   };
 
+  closeFile = (path: string): void => {
+    if (this.state.openPaths.length === 1) {
+      return;
+    }
+
+    this.setState(({ openPaths, activePath }) => {
+      const newPaths = openPaths.filter((openPath) => openPath !== path);
+
+      return {
+        activePath: path === activePath ? newPaths[0] : activePath,
```

Comment:
Instead of setting the first item as active, I'd set the item before that was just closed, as exactly how any tabs (browser, vscode, so on) works. 

For example:
https://user-images.githubusercontent.com/4838076/125816787-f8a10cd3-b588-4e1b-891d-fdfaceb97cfc.mov